### DISCUSSION TITLE
How to access Cloudflare Pages environment variable in Vike app

### DISCUSSION THREAD

**Discussion by jasonhilldm** [vikejs/vike-cloudflare#12] [Q&A]
I am using the Cloudflare adapter for Vike (vike-cloudflare) to build and deploy a small test app to Cloudflare Pages. The app uses Hono and Prisma to access a PostgreSQL database. I have stored the database URL in an environment variable in CF Pages but I don't know how to access that from within the Vike app.

Codes samples for CF Workers show this pattern for accessing the environment variables:

```ts
export interface Env {
  API_HOST: string;
}

export default {
  async fetch(request, env, ctx): Promise<Response> {
    return new Response(`API host: ${env.API_HOST}`);
  }
} satisfies ExportedHandler<Env>;
```

I am not creating a worker specifically though, so don't have that env parameter that is being passed into the fetch function above. Is there some other way to access the environment variables in CF Pages?

**Comment by brillout** [comment]
It seems to be a vike-cloudflare question @magne4000.

**Comment by magne4000** [comment]
I suggested that you take a look at https://batijs.dev/, and generate a boilerplate with Cloudflare, Drizzle, Lucia-auth, Trpc, and Hono. You will see that:
- `env` is passed in each Hono Handler and Middleware because we are using [universal-middleware](https://universal-middleware.dev/)
- `env` is always forwarded to Vike `pageContextInit`, thus making it available on Vike's [pageContext](https://vike.dev/pageContext) server-side
- Depending on where we need to use `env`, you will see some usage by `lucia-auth` middleware, and some other usage by `trpc`

You can then take inspiration of the different patterns to adapt your code.

**Reply by jasonhilldm** [reply]
Thanks...will give that a shot.
